client/state/jetpack-connection-health/reducer.js

```diff
@@ -3,52 +3,113 @@ import {
 	JETPACK_CONNECTION_HEALTHY,
 	JETPACK_CONNECTION_MAYBE_UNHEALTHY,
 	JETPACK_CONNECTION_UNHEALTHY,
+	JETPACK_CONNECTION_HEALTH_REQUEST,
+	JETPACK_CONNECTION_HEALTH_REQUEST_SUCCESS,
+	JETPACK_CONNECTION_HEALTH_REQUEST_FAILURE,
 } from 'calypso/state/action-types';
+import {
+	combineReducers,
+	withSchemaValidation,
+	keyedReducer,
+	withPersistence,
+} from 'calypso/state/utils';
+import { schema } from './schema';
 
-export const jetpackConnectionHealth = ( state = {}, action ) => {
+export const connectionHealth = withPersistence( ( state = {}, action ) => {
 	switch ( action.type ) {
 		case JETPACK_CONNECTION_HEALTHY: {
-			const { siteId } = action;
-
 			return {
 				...state,
-				[ siteId ]: {
-					...state[ siteId ],
-					jetpack_connection_problem: false,
-					is_healthy: true,
-					error: '',
-				},
+				jetpack_connection_problem: false,
+				is_healthy: true,
+				error: '',
 			};
 		}
 
 		case JETPACK_CONNECTION_MAYBE_UNHEALTHY: {
-			const { siteId } = action;
-
 			return {
 				...state,
-				[ siteId ]: {
-					...state[ siteId ],
-					jetpack_connection_problem: true,
-				},
+				jetpack_connection_problem: true,
 			};
 		}
 
 		case JETPACK_CONNECTION_UNHEALTHY: {
-			const { siteId, errorCode } = action;
+			const { errorCode } = action;
 
 			return {
 				...state,
-				[ siteId ]: {
-					...state[ siteId ],
-					jetpack_connection_problem: true,
-					is_healthy: false,
-					error: errorCode,
-				},
+				jetpack_connection_problem: true,
+				is_healthy: false,
+				error: errorCode,
 			};
 		}
 	}
 
 	return state;
-};
+} );
+
+/**
+ * `Reducer` function which handles request/response actions
+ * concerning Jetpack connection requests
+ * @param {Object} state - current state
+ * @param {Object} action - action
+ * @returns {Object} updated state
+ */
+export const requestError = withPersistence( ( state = '', action ) => {
+	switch ( action.type ) {
+		case JETPACK_CONNECTION_HEALTH_REQUEST:
+			return '';
+		case JETPACK_CONNECTION_HEALTH_REQUEST_FAILURE:
+			return action.error;
+		default:
+			return state;
+	}
+} );
+
+/**
+ * `Reducer` function which handles request/response actions
+ * concerning Jetpack connection requests
+ * @param {Object} state - current state
+ * @param {Object} action - action
+ * @returns {Object} updated state
+ */
+export const lastRequestTime = withPersistence( ( state = null, action ) => {
+	switch ( action.type ) {
+		case JETPACK_CONNECTION_HEALTH_REQUEST:
+			return action.lastRequestTime;
+		default:
+			return state;
+	}
+} );
+
+/**
+ * `Reducer` function which handles request/response actions
+ * concerning Jetpack connection requests
+ * @param {Object} state - current state
+ * @param {Object} action - action
+ * @returns {Object} updated state
+ */
+export const isLoading = withPersistence( ( state = false, action ) => {
```

Comment:
Feels off to persist the `isLoading` state. Should be fine if we end up removing it as I suggested in another comment.